core/src/vaults.ts

```diff
@@ -0,0 +1,378 @@
+    const incentiveCombinedDai = incentiveRelativeDai.plus(incentiveConstantDai);
+    return {
+        incentiveCombinedDai,
+        incentiveConstantDai,
+        incentiveRelativeDai,
+    };
+};
+
+export const fetchVaultLiquidationIncentive = memoizee(_fetchVaultLiquidationIncentive, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 3,
+});
+
+const _isVaultLiquidated = async (network: string, vault: Vault) => {
+    const contract = await getContract(network, 'MCD_DOG');
+    const typeHex = ethers.utils.formatBytes32String(vault.collateralType);
+    const eventFilter = contract.filters.Bark(typeHex, vault.address, null, null, null, null, null);
+    const liquidationEvents = await contract.queryFilter(eventFilter);
+    if (liquidationEvents.length !== 0 && vault.initialDebtDai.eq(0)) {
+        // there was a liquidation and the vault was not used again
+        const latestEvent = liquidationEvents[liquidationEvents.length - 1];
+        return {
+            isLiquidated: true,
+            liquidationDate: await fetchDateByBlockNumber(network, latestEvent.blockNumber),
+            transactionHash: latestEvent.transactionHash,
+            auctionId: new BigNumber(latestEvent.args?.id._hex).toFixed(),
+        };
+    }
+    return { isLiquidated: false };
+};
+export const isVaultLiquidated = memoizee(_isVaultLiquidated, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 2,
+});
+
+const _enrichVaultWithTransactonInformation = async (
+    network: string,
+    vault: Vault
+): Promise<VaultTransactionNotLiquidated> => {
+    const debtDai = vault.initialDebtDai.multipliedBy(vault.stabilityFeeRate);
+    const { liquidatiorContractAddress } = await fetchCollateralLiquidationLimitsAndLiquidatorAddress(
+        network,
+        vault.collateralType
+    );
+    const { incentiveCombinedDai, incentiveConstantDai, incentiveRelativeDai } = await fetchVaultLiquidationIncentive(
+        network,
+        liquidatiorContractAddress,
+        debtDai
+    );
+
+    const { liquidationRatio, oracleAddress } = await fetchLiquidationRatio(network, vault.collateralType);
+    const collateralizationRatio = vault.collateralAmount
+        .multipliedBy(vault.minUnitPrice)
+        .multipliedBy(liquidationRatio)
+        .dividedBy(debtDai)
+        .toNumber();
+    const proximityToLiquidation = collateralizationRatio - liquidationRatio;
+    const { transactionFeeLiquidationEth, transactionFeeLiquidationDai } = await getApproximateLiquidationFees(
+        network
+    );
+    const osmPrices = await getOsmPrices(network, oracleAddress);
+    const { nextUnitPrice, nextPriceChange, currentUnitPrice } = osmPrices
+        ? osmPrices
+        : {
+              nextUnitPrice: new BigNumber(NaN),
+              currentUnitPrice: new BigNumber(NaN),
+              nextPriceChange: undefined,
+          };
+    const state = proximityToLiquidation < 0 ? 'liquidatable' : 'not-liquidatable';
```

Comment:
This logic doesn't account for the limits, right? If the limits are reached already, the transaction wouldn't go trough. If it will be reached after the execution, it will be done partially (so maybe the `incentiveRelativeDai` will be affected?)